### PR TITLE
Add no-nested-try-catch lint rule

### PR DESCRIPTION
**Pull request by danielchen0**
## Summary
- Adds `no-nested-try-catch` rule that flags try-catch blocks nested inside other try/catch/finally blocks
- Suggests extracting inner try-catch to separate functions

## Test plan
- [x] All existing tests pass
- [x] 7 new tests pass
- [x] TypeScript compiles
- [x] Prettier passes

### CODE REVIEW

File: src/rules/no-nested-try-catch.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import type { LintResult } from '../types';
+
+const RULE_NAME = 'no-nested-try-catch';
+
+export function noNestedTryCatch(ast: File, _code: string): LintResult[] {
+  const results: LintResult[] = [];
+
+  traverse(ast, {
+    TryStatement(path) {
+      // Check if any ancestor is also a TryStatement's block
```
Comment (arnavsurve) [inline src/rules/no-nested-try-catch.ts]: Nit: The comment says "Check if any ancestor is also a TryStatement's block" but the code checks for any `TryStatement` ancestor generally (not specifically the `.block` child). The logic is correct regardless -- just a minor wording thing. Could be simplified to "Check if any ancestor is a TryStatement" to match what the code actually does.

**Review by arnavsurve — APPROVED** [review]
Clean, well-structured addition. The rule implementation follows established patterns (matches `no-type-assertion.ts` style), tests are thorough (7 cases covering try/catch/finally nesting, deep nesting, sibling try-catch, and the function boundary edge case), and the README documentation is properly updated.

One minor observation: the ancestor walk doesn't stop at function boundaries, so `try { const fn = () => { try { ... } catch {} }; } catch {}` will be flagged. The test explicitly documents this as intentional/acceptable, which I agree with -- it nudges users toward extracting the inner function outside the try block entirely, which is the cleaner pattern.

The comment on line 12 says "Check if any ancestor is also a TryStatement's block" but the code checks for `TryStatement` ancestors generally (not just the `block` property). The logic is still correct since any try nested anywhere within a TryStatement node (block, handler, or finalizer) should be flagged. The comment could be slightly more precise, but this is a pure nit.

LGTM.

🤖 Generated with [Claude Code](https://claude.com/claude-code)
